model/tasks/ImportAndCompile.php

```diff
@@ -121,88 +180,78 @@ public function jsonSerialize()
 
     /**
      * @param array $params
-     * @throws \common_exception_MissingParameter
-     * @throws \common_exception_InconsistentData
+     *
+     * @throws InvalidServiceManagerException
+     * @throws InconsistentDataException
+     * @throws MissingParameterException
      */
-    protected function checkParams(array $params)
+    private function checkParams(array $params): void
     {
-        if (!isset($params[self::OPTION_FILE])) {
-            throw new \common_exception_MissingParameter('Missing parameter `' . self::OPTION_FILE . '` in ' . self::class);
-        }
-        if (!isset($params[self::OPTION_IMPORTER])) {
-            throw new \common_exception_MissingParameter('Missing parameter `' . self::OPTION_IMPORTER . '` in ' . self::class);
+        foreach ([self::OPTION_FILE, self::OPTION_IMPORTER] as $param) {
+            if (!isset($params[$param])) {
+                throw new MissingParameterException(sprintf(
+                    'Missing parameter `%s` in %s',
+                    $param,
+                    self::class
+                ));
+            }
         }
 
         $importer = $this->getImporter($params[self::OPTION_IMPORTER]);
 
         if (!$importer instanceof AbstractTestImporter) {
-            throw new \common_exception_InconsistentData('Wrong importer `' . $params[self::OPTION_IMPORTER] . '`');
+            throw new InconsistentDataException(sprintf(
+                'Wrong importer `%s`',
+                $params[self::OPTION_IMPORTER]
+            ));
         }
     }
 
     /**
-     * @param string $classLabel
-     * @return \core_kernel_classes_Class
+     * @param array $classLabels
+     *
+     * @return CoreClass
      */
-    protected function checkSubClasses($classLabel = '')
+    private function checkSubClasses(array $classLabels = []): CoreClass
     {
-        $parent = new \core_kernel_classes_Class(DeliveryAssemblyService::CLASS_URI);
-        if (!$classLabel) {
+        $parent = new CoreClass(DeliveryAssemblyService::CLASS_URI);
+
+        if (empty($classLabels)) {
             return $parent;
-        }
-        $deliveryClasses = $parent->getSubClasses(true);
-        $class = null;
-        if ($classLabel) {
-            foreach ($deliveryClasses as $deliveryClass) {
-                if ($classLabel === $deliveryClass->getLabel()) {
+        } else {
+            $deliveryClasses = $parent->getSubClasses(true);
+
+            foreach (array_values($deliveryClasses) as $index => $deliveryClass) {
+                if (isset($classLabels[$index]) && $classLabels[$index] === $deliveryClass->getLabel()) {
                     $class = $deliveryClass;
+                } else {
```

Comment:
This is bizarre to me. Is it correct the `else`  -> `break` here?

Reply:
I found some mistakes, so I published a [new "version"](https://github.com/oat-sa/extension-tao-delivery-rdf/pull/360/files#diff-fcb46d0faf618bff44e3b90cc5a38392R234) :)
Please, check it.